Dockerfile

```diff
@@ -104,13 +110,20 @@ COPY --from=cuda-12 dist/lib/ollama/cuda_v12 /lib/ollama/cuda_v12
 COPY --from=jetpack-5 dist/lib/ollama/cuda_v11 lib/ollama/cuda_jetpack5
 COPY --from=jetpack-6 dist/lib/ollama/cuda_v12 lib/ollama/cuda_jetpack6
 
-FROM --platform=linux/arm64 scratch AS rocm
+FROM scratch AS rocm
 COPY --from=rocm-6 dist/lib/ollama/rocm /lib/ollama/rocm
 
 FROM ${FLAVOR} AS archive
 COPY --from=cpu dist/lib/ollama /lib/ollama
 COPY --from=build /bin/ollama /bin/ollama
 
+FROM --platform=linux/amd64 amd64 AS dist-amd64
+COPY --from=rocm /lib/ollama /lib/ollama
+FROM --platform=linux/arm64 arm64 AS dist-arm64
+FROM dist-$TARGETARCH AS unified-archive
+COPY --from=cpu dist/lib/ollama /lib/ollama
+COPY --from=build /bin/ollama /bin/ollama
```

Comment:
This shouldn't be necessary since archive output can be assembled from distinct targets while the docker output doesn't require merging stages

Reply:
Can you elaborate?  I'm not seeing how to invoke `docker buildx build ...` once and glue together the two different FLAVORs to get cuda+rocm builds at once.  I want to avoid invoking build multiple times since if you touch any files in the local repo while the first build is running, the second build will get potentially different content.

Reply:
Given build_docker does require two steps, I'll just switch to that model for build_linux as well.